### DISCUSSION TITLE
Attempt to access a private repository reports 404

### DISCUSSION THREAD

**Discussion by DG12** [community/community#35562] [Repositories]
If you are not signed  in to GitHub attempting to access a private repository  presents a cute but unhelpful 404. 

Prospective users or users who were referred to a repository are immediately frustrated.

Surely  there is a more helpful page that can be presented.

Something that doesn't suggest. "Find code, projects, and people on GitHub:" or

"Contact Support - GitHub Status - @githubstatus"  (shouldn't those be comma separated?)

**Comment by airtower-luna** [comment]
> If you are not signed in to GitHub attempting to access any repository presents a cute but unhelpful 404.

That will happen only for *private* repositories, which you can't access without logging in. For public repositories you should see a repository listing as if you were logged in (minus actions that'd require logging in).

**Reply by DG12** [reply]
Thank you for replying I thought that I had tried a public repository and got the same response, but I guess not.

Even for private repositories this response if very user Unfriendly and not helpful.

**Reply by airtower-luna** [reply]
> Even for private repositories this response if very user Unfriendly and not helpful.

That's kind of the point. :sweat_smile: A "you have to log in to see this repository" message or similar would let anyone know that a repository of the name exists, and possibly a certain file within the repository, too. Which the repository owner might not want strangers to know.

Using 404 instead of 401/403 is a common pattern to make it impossible for unauthenticated clients to tell the difference between paths that don't exist and that they don't have access to.

**Comment by DG12** [comment]
I am very familiar with the security consideration of "if they even know it exists that provides an attack vector".

A much better response would be:
 "Either the page : xxxx does not exist or it is private and requires a logged in user to have access."

This would not disclose the existence or restriction of the page.

PS I have changed the title to be more correct.
